### PR TITLE
[xenial] Target both Trusty and Xenial for staging tests

### LINKED ISSUE #3966: [xenial] Support running CI staging tests against both 14.04 and 16.04
To better detect bugs during the transition period from Ubuntu 14.04 to 16.04, we should run two staging workflows in CI until the transition is complete: a Trusty workflow and a Xenial workflow.

Investigate whether per-PR CI performance hit is acceptable, otherwise configure to run one of the two (initially Xenial) nightly.

Part of #3204.

### PR DESCRIPTION
**Pull request by emkll**
## Status

Ready for review

## Description of Changes

Fixes #3966 

Introduces nightly staging xenial runs in CI for automated infra testing.

## Testing

- [ ] CI run with Xenial machines should occurs nightly at 0:00 UTC
- [ ] CI run completes without failure
- [ ] Other CI features are unaffected

## Deployment

1. Dev-env only

### If you made non-trivial code changes:

- [X] I have written a test plan and validated it for this PR

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -236,3 +265,14 @@ workflows:
...
+    triggers:
+      - schedule:
+          cron: "0 0 * * *"
+          filters:
+            branches:
+              only:
+                - develop
+    jobs:
+      - static-analysis-and-no-known-cves
+      - staging-test-with-rebase-xenial`
```
Comment (redshiftzero) [inline .circleci/config.yml]: is the backtick at the end of `staging-test-with-rebase-xenial` a typo?

Comment (emkll) [reply .circleci/config.yml]: it is !

Comment (emkll) [reply .circleci/config.yml]: fixed, good catch !

**Comment by kushaldas** [comment]
Kicking the failed `admin-tests`

**Pull request merged**